test_ghstack.py

```diff
@@ -2081,6 +2081,49 @@ def test_default_branch_change(self) -> None:
 rUP1 Commit 1
 rINI0 Initial commit''')
 
+    # ------------------------------------------------------------------------- #
+
+    def test_gpgsign_should_fail_if_no_default_key(self) -> None:
+        print("####################")
+        print("### test_gpgsign_should_fail_if_no_default_key")
+        print("###")
+
+        @contextlib.contextmanager
+        def signing_enabled() -> Iterator[None]:
+            # Create a config file that enables signing
+            with tempfile.NamedTemporaryFile(delete=False) as f:
+                f.writelines(f"{x}\n".encode() for x in [
+                    "[commit]",
+                    "\tgpgsign=true",
+                ])
+            # Set the env var
+            config = os.environ.get("GIT_CONFIG", None)
+            os.environ["GIT_CONFIG"] = f.name
```

Comment:
Modifying os.environ is generally frowned upon because it is global state that affects the rest of the process. So it would be better if os.environ was as an explicit argument to main(), so that we can more easily replace its contents when testing (as is the case here). Unfortunately, this is not the case, and I shouldn't really be asking you to do the free work of refactoring all of those sites haha. So I'll probably just take this as is.